### ISSUE TITLE
Wooden storage crates deleting inventory when picked up and replaced

### ISSUE THREAD

**Issue by Hazmatik** [BluSunrize/ImmersiveEngineering#2245]
Version: ImmersiveEngineering-0.10-61

It doesn't always happen but sometimes the wooden storage crates after I break them and place them down elsewhere are empty. This is not the same thing as the visual bug others have reported as the items aren't just invisible they are completely gone.

**Comment by MJRLegends** [comment]
@BluSunrize Just to add more information to this, the mod pack is Space Astronomy 2 and is version v1.0.9 Beta of the pack. And its a 1.10.2 pack. 

If you need any more information please let me know, i think this was tested on a sponge server, so @Hazmatik might need to confirm he can get this issue on a single player world

**Comment by MJRLegends** [comment]
@BluSunrize Just to add more information to this, the mod pack is Space Astronomy 2 and is version v1.0.9 Beta of the pack. And its a 1.10.2 pack. And its my mod pack

If you need any more information please let me know, i think this was tested on a sponge server, so @Hazmatik might need to confirm he can get this issue on a single player world

**Comment by MJRLegends** [comment]
@BluSunrize im willing to test it more and see what i can find out to the cause and if its due to anything else, please let me know if that is needed.

**Comment by BrianEdmonds** [comment]
I've seen this once recently myself, but have not reproduced, so didn't
mention anything.

**Comment by mindforger** [comment]
if i remember correct, but my information could be outdated ... the crate stores it's content in NBT data, when an item with excessive NBT contents is added it may corrupt the crate data (there is most probably an error report on the server log the moment you break it)

chances are high that only specific items or specific item combinations are causing this or when the NBT tag is getting too big

you can take that into account while testing :)

but my informations could be totally wrong

**Comment by BrianEdmonds** [comment]
The crates I picked up were from an engineer's house in a village.  I'd
not added or removed anything, so it's unlikely it was due to excess NBT
data.  I did get disconnected from the server and relogged between at
some point between grabbing them and putting them down back home to use
them, but I wouldn't have expected that to empty them.

**Comment by Hazmatik** [comment]
Yeah same here the ones I picked up were from an engineers house in a village, i looked in them, picked them up, placed them back down at home and the contents were gone. I didn't disconnect at all between time, either.

**Comment by BluSunrize** [comment]
Huh. Is this in singleplayer or on a server?

I wonder if a simple markDirty() when reading the loottable would work...

**Comment by Hazmatik** [comment]
It happened to me on a Sponge server I haven't tested in singleplayer, and out of the 3 crates at the engineers house it was only one that kept it's inventory so it isn't a guarantee that it could be reproduced.

**Comment by BluSunrize** [comment]
Sponge, eh? That might have factored into the problem.
Could you do me a solid and see if you can reproduce it in a singleplayer world? Even if you only get it to happen once in ~20 crates, that would give me confirmation that it'S a legitimate error outside of Sponge.

**Comment by BrianEdmonds** [comment]
Regular Forge server for me. Custom modpack with IE as the primary mod,
plus some incidentals like Tomb Many Graves, Iron Backpacks, and such to
make play less grindy.

**Comment by BluSunrize** [comment]
I'll include an experimental fix for this in the next release. Feel free to give it another go after that is released (hopefully later tonight)

**Comment by MJRLegends** [comment]
@BluSunrize if that fix does work would it be possible to please get it for MC 1.10.2 version as well as MC 1.11.2?

**Comment by BluSunrize** [comment]
I'm not planning to release any more 1.10 builds I'm afraid.
Plus, that fix isn't even confirmed to work.

**Comment by MJRLegends** [comment]
Ok no problem, i will just have to remove them out my pack by removing the recipe and hiding it from jei

**Comment by BrianEdmonds** [comment]
> Ok no problem, i will just have to remove them out my pack

To be clear, the problem was not with crates losing inventory
regularly.  The only instance I've seen was after picking some up from
the engineer's village house.  They performed entirely normally after
that initial glitch.

**Comment by mindforger** [comment]
@BrianEdmonds this speaks for the fix Blu tried for this
but the Bug that @MJRLegends expired looks more intense and reproducible, i recommend @MJRLegends and @Hazmatik to check this with Items like Tinker Tools, Chargable Tools, Wands and other Stuff that has "Informations" Stored in it ... at least we can maybe find somethign reproducible this way and report it to sponge probably :D

**Comment by Hazmatik** [comment]
As far as things that store information stored in it the crates from a villagers house only thing i could think that might have NBT associated with it would be the engineers blueprints maybe but I'm not sure about that. I will see if I can reproduce it on a singleplayer world with stuff that does have info stored on it.

**Comment by malte0811** [comment]
I'm going to do some testing with large NBT tag compounds now, even though I don't think that would cause this. I worked with large NBT tag compounds recently for a Forge PR and it seems like the only place where the NBT size is limited is when it is sent over the network. Recursion is also limited, but I think any user would notice if they nested 500+ bags/crates. Has anyone checked for errors in the server logs when the non-village-crate bug is happening?

**Comment by malte0811** [comment]
Test results: No item loss so far. I have used some quite large tags for this (something like 9 items with 3^8 doubles each, as a tree (Ok, NBT is always a tree, but you know what I mean)), so I don't think large NBT tags are the cause of this. I have found a dupe bug on item pickup though, it only happens when there is a LOT of NBT in the player inventory.

**Comment by MJRLegends** [comment]
Any more news on this issue?

**Comment by BluSunrize** [comment]
Well I already said I wasn't doing any more 1.10 builds, plus I never got anyone to test it without Sponge, so no.

**Comment by MJRLegends** [comment]
Can i disable them from spawning or remove them in any way @BluSunrize

**Comment by VallenFrostweaver** [comment]
I have experienced this on the Raxpack 2 modpack for 1.11.2. :/

**Comment by underr** [comment]
Experiencing this issue on the Radon modpack for 1.12.

**Comment by labr940817** [comment]
Experiencing this in 1.12.1 single player Immersive Engineering only :(

**Comment by BluSunrize** [comment]
Versionnumbers?

**Comment by MJRLegends** [comment]
Not sure on certain versions but the issue seems to be from MC 1.10.2 and 1.11.2 and 1.12 and 1.12.1, I'd say its a issue in all latest versions unless you have fixed the issue in newer versions....

**Comment by BluSunrize** [comment]
"Latest" is not a valid answer. The contribution guidelines exist for a reason. If someone can *replicate* this reliably, on an instance with just Forge and IE, and provide version numbers, then I can actually look into it .-.

**Comment by MJRLegends** [comment]
Yeah i can reproduce it 100% of the time
MC: 1.10.2 
Forge: 12.18.3.2422
Immersive Engineering: 0.10 - 61

Since you ported the mod to latest mc versions the bug will be the same in all MC versions from 1.10.2 so version numbers shouldn't matter as much for them since the error will show in each version

**Comment by MJRLegends** [comment]
Also a quick google search for the pack @VallenFrostweaver and @underr are using is the following:
MC: 1.12.1
Forge: 14.22.0.2464
Immersive Engineering: 0.12 - 71

The mod pack is public so downloading it shows what version they are using since they might not respond to your message so this is a way to get the version without them needing to reply

**Comment by MJRLegends** [comment]
And i have no doubt that @labr940817 is using the same as above for the MC 1.12 pack hes playing too

**Comment by MJRLegends** [comment]
@BluSunrize Also a quick google search for the pack @VallenFrostweaver are using is the following:
MC: 1.11.2
Forge: 13.20.1.2393
Immersive Engineering: 0.11 - 63

The mod pack is public so downloading it shows what version they are using since they might not respond to your message so this is a way to get the version without them needing to reply

i dont think versions matter here the bug is in all the MC 1.10.2 and 1.11.2 and 1.12/1.12.1 builds

**Comment by BluSunrize** [comment]
Uh no, not really. I have made changes to those dropsystems since. That's why I want someone to present a properly documented testcase on build 0.12-71

**Comment by labr940817** [comment]
IE 0.12-71
MC 1.12.1
Forge:    1.12.1-forge1.12.1-14.22.0.2474

Sorry i'm late, hope this helps. I think that it's caused by me having like 5 or so crates filled with items in my player inventory. I guess there's a limit :p

**Comment by royalcw** [comment]
Experienced this multiple times now with the same three crates I have.  Lost the items roughly 40% of the time when I pick them up.  It seems to happen when I break multiple crates in a row and pick them all up at once.  If I just break a single crate with stuff in it and pick it up then the items stay in the crate.
IE Version 0.12 - 71
MC 1.12.1
Forge: 14.22.0.2474

**Comment by kMaiSmith** [comment]
IE - 0.12 - 71
MC - 1.12.1
Forge - 14.22.0.2469
Modpack - FTB Unstable 1.12 version 2.0.3

In addition to the crate pickup and placement issue, I am also noticing a similar issue with the toolbox (i lost all my ie stuff early on :( ) and the crusher appears to have just randomly lost its inventory mid-processing.  Since these issues are inventory related I figured i'd post a comment on this issue rather than open a new one.  

Toolbox:

I opened the toolbox in my inventory and filled it with stuff, when i closed it and re-opened it, everything was gone.  I was able to reproduce this 100% of the time.  When i place the toolbox on the ground and placed items in it, the items persist even after picking the toolbox up, but as soon as i place a new item in the toolbox from my inventory and close the toolbox, all of the items are deleted. 

Crusher:

Not really much terribly interesting regarding the crusher.  I threw a stack of yellowrite ore, a half stack of gold, a half stack of iron, and a half stack of Thermal Expansion lead in the crusher and walked away.  When i came back the crusher had processed some of the stuff, but most of the resulting dusts were missing.
